geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxyStats.java

```diff
@@ -313,19 +332,38 @@ public void incCqCount() {
     this._stats.incInt(_cqCountId, 1);
   }
 
+  /**
+   * Increments the "numberThreadsPutInQueue" stat.
+   */
+  public void incNumberThreadsPutInQueue() {
+    synchronized (this) {
```

Comment:
CacheClientProxyStats are atomic so you should not need synchronization (I saw it in three places: inc, dec, and set).

Reply:
Thanks for all the comments. I will update solution.